oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3Ds.java

```diff
@@ -89,11 +133,98 @@ public void setUp() throws Exception {
             String.valueOf(randomGen.nextInt(9999)) + "-" + String.valueOf(randomGen.nextInt(9999))
                 + "-s3ds-unittest-autogenerated";
         createdBucketNames.add(bucket);
-        props.setProperty(S3Constants.S3_BUCKET, bucket);
-        props.setProperty("secret", "123456");
         super.setUp();
     }
 
+    @Test
+    public void testInitiateDirectUploadUnlimitedURIs() throws DataRecordUploadException,
+            RepositoryException {
+        ConfigurableDataRecordAccessProvider ds
+                  = (ConfigurableDataRecordAccessProvider) createDataStore();
+        long uploadSize = ONE_GB * 50;
+        int expectedNumURIs = 5000;
+        DataRecordUpload upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+
+        uploadSize = ONE_GB * 100;
+        expectedNumURIs = 10000;
+        upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+
+        uploadSize = ONE_GB * 200;
+        upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+    }
+
+    @Test
+    public void testGetDownloadURIIT() throws IOException, RepositoryException {
+        DataRecord record = null;
+        DataRecordAccessProvider ds = (DataRecordAccessProvider) createDataStore();
+        InputStream testStream = randomStream(0, 256);
+        record = doSynchronousAddRecord(ds, testStream);
+        URI uri = ((DataRecordAccessProvider) ds).getDownloadURI(record.getIdentifier(), DataRecordDownloadOptions.DEFAULT);
+        Assert.assertNotNull("uri is null", uri);
+    }
```

Comment:
Unless I am misreading, this method and below look exactly same to me. Is there a reason it is names as *IT? If same then delete this.